### PR TITLE
rate: add prometheus definitions, docs, and clearer names

### PR DESCRIPTION
**Pull request by boxofrad**
### Description
While testing the server-side rate limiting work, I spotted that we were missing docs and a Prometheus definition for the counter that is incremented whenever a limit is exceeded. Also renamed the counter for better clarity/consistency.

### CODE REVIEW

File: website/content/docs/agent/telemetry.mdx [hunk 1]
```diff
@@ -477,6 +477,7 @@ These metrics are used to monitor the health of the Consul servers.
...
+| `consul.rpc.rate_limit.exceeded`                    | Increments whenever an RPC is over a configured rate limit. Note: in permissive mode, the RPC will have still been allowed to proceed.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | RPCs                              | counter |
```
Comment (trujillo-adam) [inline website/content/docs/agent/telemetry.mdx:480]:
```suggestion
| `consul.rpc.rate_limit.exceeded`                    | Increments whenever an RPC is over a configured rate limit. When Consul is in `permissive` mode, the RPC is allowed to proceed.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | RPCs                              | counter |
```
Actually, what is in permissive mode in this context? I assumed that `permissive` is a value that you you can define in some configuration, but I'm not sure. Also, we should just use simple present tense per the style guide.

**Review by trujillo-adam — COMMENTED** [review]
A few questions and suggestions for consistency with the style guide.

File: website/content/docs/agent/telemetry.mdx (hunk 1 above)
Comment (boxofrad) [reply website/content/docs/agent/telemetry.mdx:480]: Thanks for fixing my terrible writing! 😆 ❤️  

In this context, the rate limiter is in permissive mode, and there can be multiple limiters involved in a single RPC (e.g. global, ip-based, tenant-based) which are configured independently.

**Review by jmurret — APPROVED** [review]
👍 🎉

**Pull request merged**